### PR TITLE
fix: dragonfly_connection should only access the original reply_builder

### PR DESCRIPTION
**Pull request by romange**
ConnectionContext.reply_builder can be injected and replaced by the service logic.
before - dragonfly_connection accessed it via cc_->reply_builder in some places,
which led it to access the injected object. Moreover, EVAL commands can be offloaded
to another thread and that thread could inject the object, making the access to cc_->reply_builder_
non thread-safe.

Now dragonfly_connection copies aside the replier_builder_ pointer, and uses only this pointer for communicating with client.

Also, remove redundant arguments.

### CODE REVIEW

File: src/facade/dragonfly_connection.cc [hunk 1]
```diff
@@ -1212,24 +1214,22 @@ void Connection::HandleMigrateRequest() {
...
     // which can never trigger since we Joined on the dispatch_fb_ above and we are
     // atomic in respect to our proactor meaning that no other fiber will
     // launch the DispatchFiber.
     if (!this->Migrate(dest)) {
       return;
     }
   }
-
-  // In case we Yield()ed in Migrate() above, dispatch_fb_ might have been started.
-  LaunchDispatchFiberIfNeeded();
```
Comment (adiholden) [inline src/facade/dragonfly_connection.cc:1224]: why did you remove this?

Comment (romange) [reply src/facade/dragonfly_connection.cc:1224]: we do that inside `OnPostMigrateThread` function.

**Review by adiholden — APPROVED** [review]

**Review by adiholden — APPROVED** [review]
